### PR TITLE
Fix uptime on AIX systems when less than 24 hours

### LINKED ISSUE #61557: [BUG] status.uptime will fail if uptime less than 24 hours (AIX minions only)
**Description**
If uptime on AIX minion is less than 24 hours call to `status.uptime` will generate `IndexError`.

**Setup**
AIX 7.2 minion with `uptime` less than 24 hours.

- [x] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

  
**Steps to Reproduce the behavior**
Issue `salt AIX_MINION status.uptime` from salt master.

**Expected behavior**
Information regarding uptime appears on stdout.

**Actual behaviour**
No information is returned, instead an error appears:
``` 
    The minion function caused an exception: Traceback (most recent call last):
      File "/opt/salt/lib/python2.7/site-packages/salt/minion.py", line 1606, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/opt/salt/lib/python2.7/site-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/opt/salt/lib/python2.7/site-packages/salt/modules/status.py", line 243, in uptime
        seconds = _get_boot_time_aix()
      File "/opt/salt/lib/python2.7/site-packages/salt/modules/status.py", line 92, in _get_boot_time_aix
        hms = days[1].split(':')
    IndexError: list index out of range
``` 

**Versions Report**
Salt master running at version 3002.6, minion running at version 2018.3.2.

**Additional context**
If uptime of AIX system is less than 24 hours the output of `ps -o etime= -p 1` is as follows:
``` 
$ ps -o etime= -p 1
   11:10:06
``` 
This is how the output looks on a host with 96+ days of uptime (for comparison):
``` 
$ ps -o etime= -p 1
96-20:58:27
```

### PR DESCRIPTION
**Pull request by OrangeDog**
### What does this PR do?
Fixes `status.uptime` on AIX when less than 24 hours.

### What issues does this PR fix or reference?
Fixes: #61557

### Previous Behavior
Exception raised

### New Behavior
Correct result

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### CODE REVIEW

**Comment by dmurphy18** [comment]
@OrangeDog Next time assign yourself the issue too, as I just worked up a solution but I still have to write tests, so will go with yours.  Wondering what the format is for a system that has been up over a year. SaltStack only gets loaner AIX systems for at most a month,

**Comment by dmurphy18** [comment]
re-run centos-8

**Comment by OrangeDog** [comment]
@dmurphy18 ah sorry. I'd meant to put you here, not there, as you were the original author.

File: salt/modules/status.py [hunk 1]
```diff
@@ -69,21 +69,21 @@ def _get_boot_time_aix():
...
-    t is 7-20:46:46
     """
-    boot_secs = 0
     res = __salt__["cmd.run_all"]("ps -o etime= -p 1")
     if res["retcode"] > 0:
         raise CommandExecutionError("Unable to find boot_time for pid 1.")
     bt_time = res["stdout"]
-    days = bt_time.split("-")
-    hms = days[1].split(":")
+    match = re.match(r"(?:(\d+)-)?(?:(\d\d):)?(\d\d):(\d\d)", bt_time)
```
Comment (jose1711) [inline salt/modules/status.py:77]: Could you maybe consider using named capture groups? Ex:
```
re.match(r"(?:(?P<days>\d+)-)?((?P<hours>\d\d):)?(?P<minutes>\d\d):(?P<seconds>\d\d)", '7-20:46:46').group('hours')
```

**Comment by dmurphy18** [comment]
@OrangeDog No problem, glad to see you contribute and on AIX of all things :)

**Comment by Ch3LL** [comment]
@dmurphy18 can you review the PR please?

**Review by Ch3LL — DISMISSED** [review]

**Review by dmurphy18 — CHANGES_REQUESTED** [review]
Code does now work on live AIX since match expression does not allow for spaces in bt_time

File: salt/modules/status.py (hunk 1 above)
Comment (OrangeDog) [reply salt/modules/status.py]: Could do, but it seemed a bit overkill.

File: tests/pytests/unit/modules/test_status.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    mock.return_value = {"stdout": "foo", "retcode": 0}
+    with patch.dict(status.__salt__, {"cmd.run_all": mock}):
+        with pytest.raises(CommandExecutionError):
+            status._get_boot_time_aix()
+
+    mock.return_value = {"stdout": "", "retcode": 1}
+    with patch.dict(status.__salt__, {"cmd.run_all": mock}):
+        with pytest.raises(CommandExecutionError):
+            status._get_boot_time_aix()
```
[the remark this answers is not in the document]
Comment (dmurphy18) [reply tests/pytests/unit/modules/test_status.py:43]: Thanks, that worked on a live AIX 7.2

**Review by dmurphy18 — APPROVED** [review]
Thanks for the fix and adding that condition to the tests too

**Comment by dmurphy18** [comment]
re-run windows

**Comment by dmurphy18** [comment]
re-run all

File: salt/modules/status.py (hunk 1 above)
Comment (dmurphy18) [reply salt/modules/status.py]: fine as is

**Pull request merged**